### ISSUE TITLE
think about how the new (auto)tagging of history objects relates to version tags in history module

### ISSUE THREAD

**Issue by sennierer** [acdh-oeaw/apis-core-rdf#1117]
if #1074 is merged collections can be activated to keep all new versions created in one collection. Given that giving version tags for every single entity might be a lot of work we could think of using the collections instead of the version tag.

**Comment by sennierer** [comment]
Main feature I have in mind is allowing projects to "publish" certain versions of their objects. If we implement that using the collections we might want to remove the tags from history.

**Comment by b1rger** [comment]
> Main feature I have in mind is allowing projects to "publish" certain versions of their objects. If we implement that using the collections we might want to remove the tags from history.

Technically, this is already possible - we can add historic entities to a collection, we can also create views that only display model instances that are in a a collection. So the main question probably is: what should the interface look like?

**Comment by sennierer** [comment]
>  what should the interface look like?
I think in the history view there should be a possibility to add a certain version to a collection. And when this version is added, all relations valid at that time should also be added (potentially we could allow to (de)select that behavior).

The browsing views could be just filtered versions of the history entities views. The detail view of the history object needs to be extended for the version objects of the relations valid at that time.

Might be worth to discuss whether collections should get a boolean `history_collection` that indicates that it is used for grouping history objects to a version (this would allow to only allow history objects to get saved in the collection, filter out that collections in the other forms etc).

**Comment by sennierer** [comment]
Just created #1120 for updating the history detail view to include relations

**Comment by b1rger** [comment]
> > what should the interface look like?
> 
> I think in the history view there should be a possibility to add a certain version to a collection. And when this version is added, all relations valid at that time should also be added (potentially we could allow to (de)select that behavior).

"all relations" meaning all relations connected to this one instance?


> Might be worth to discuss whether collections should get a boolean `history_collection` that indicates that it is used for grouping history objects to a version (this would allow to only allow history objects to get saved in the collection, filter out that collections in the other forms etc).

We can put collections in other collections, so we can simply provide an option to pass a `history_collection` to the form field and it is then only possible to use collections in this `history_collection` for tagging

**Comment by sennierer** [comment]
> "all relations" meaning all relations connected to this one instance?

yes, all relations that existed by the point in time and not the relation itself, but the history object of the relation

> We can put collections in other collections, so we can simply provide an option to pass a history_collection to the form field and it is then only possible to use collections in this history_collection for tagging

yes, forgot about that

**Comment by sennierer** [comment]
decided to remove the tag from history and use collections isntead
